cmake/TestSetup.cmake

```diff
@@ -152,7 +152,7 @@ function(setup_test_parallel nproc dir arg extra_labels)
     list(APPEND labels "${extra_labels}")
 
     if(Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP)
-      set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-num-devices=${NUM_GPU_DEVICES_PER_NODE}")
+      set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-map-device-id-by=mpi_rank")
```

Comment:
This addresses https://github.com/parthenon-hpc-lab/parthenon/pull/934 to 0th order (not removing/cleaning up the `NUM_GPU_DEVICES_PER_NODE` var).

Reply:
Can we close that one?